src/Core/Grid/Definition/Factory/CmsPageDefinitionFactory.php

```diff
@@ -0,0 +1,330 @@
+                ->setOptions([
+                    'field' => 'head_seo_title',
+                ])
+            )
+            ->add((new DataColumn('position'))
+                ->setName($this->trans('Position', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'position',
+                ])
+            )
+            ->add((new ToggleColumn('active'))
+                ->setName($this->trans('Displayed', [], 'Admin.Global'))
+                ->setOptions([
+                    'field' => 'active',
+                    'route' => 'admin_cms_pages_toggle',
+                    'primary_field' => 'id_cms',
+                    'route_param_name' => 'cmsId',
+                ])
+            )
+            ->add((new ActionColumn('actions'))
+                ->setName($this->trans('Actions', [], 'Admin.Global'))
+                ->setOptions([
+                    'actions' => (new RowActionCollection())
+                        ->add((new LinkRowAction('edit'))
+                            ->setName($this->trans('Edit', [], 'Admin.Actions'))
+                            ->setIcon('edit')
+                            ->setOptions([
+                                'route' => 'admin_cms_pages_edit',
+                                'route_param_name' => 'cmsId',
+                                'route_param_field' => 'id_cms',
+                            ])
+                        )
+                        ->add((new SubmitRowAction('delete'))
+                            ->setName($this->trans('Delete', [], 'Admin.Actions'))
+                            ->setIcon('delete')
+                            ->setOptions([
+                                'method' => 'DELETE',
+                                'route' => 'admin_cms_pages_delete',
+                                'route_param_name' => 'cmsId',
+                                'route_param_field' => 'id_cms',
+                                'confirm_message' => $this->trans(
+                                    'Delete selected item?',
+                                    [],
+                                    'Admin.Notifications.Warning'
+                                ),
+                            ])
+                        ),
+                ])
+            )
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getFilters()
+    {
+        $actionsTypeOptions = [
+            'reset_route' => 'admin_common_reset_search_by_filter_id',
+            'reset_route_params' => [
+                'filterId' => self::GRID_ID,
+            ],
+            'redirect_route' => 'admin_cms_pages_index',
+        ];
+
+        if ($this->cmsCategoryParentId) {
+            $actionsTypeOptions['redirect_route_params'] = [
+                'id_cms_category' => $this->cmsCategoryParentId,
+            ];
+
+            $actionsTypeOptions['reset_route_params']['id_cms_category'] = $this->cmsCategoryParentId;
```

Comment:
Same question here, but I think I got it, you need to pass the parameter for the redirection route right?

Reply:
yep :+1:

Reply:
Well it's great that this works without creating another reset controller, I didn't imagine this generic controller could manage more advanced use cases ^^